src/main/scala/org/ergoplatform/network/ErgoSyncTracker.scala

```diff
@@ -98,21 +114,43 @@ final case class ErgoSyncTracker(system: ActorSystem,
     }
   }
 
-  protected[network] def outdatedPeers: IndexedSeq[ConnectedPeer] = {
+  /**
+    * Helper method to clear statuses of peers not updated for long enough
+    */
+  private[network] def clearOldStatuses(): Unit = {
+    val currentTime = timeProvider.time()
+    val peersToClear = statuses.filter { case (_, status) =>
+      status.lastSyncSentTime.exists(syncTime => (currentTime - syncTime).millis > ClearThreshold)
+    }.keys
+    if (peersToClear.nonEmpty) {
+      val keysToRemove = peersToClear.toVector
+      log.debug(s"Clearing stalled statuses for $keysToRemove")
+      // we set status to `Unknown` and reset peer's height
+      keysToRemove.foreach(p => updateStatus(p, Unknown, None))
```

Comment:
I guess no need for calling `peersToClear.toVector` ... `keysToRemove` is also confusing name

Reply:
Eliminated `keysToRemove`